src/util-fopen.c

```diff
@@ -0,0 +1,54 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Victor Julien <vjulien@oisf.net>
+ *
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "util-debug.h"
+#include "util-fopen.h"
+
+/** \brief fopen like function to open with explict permissions
+ *  \note permissions in mode will be limited by process wide `umask` settings
+ */
+FILE *SCFopen(const char *path, const char *m, mode_t mode)
```

Comment:
Can we add more documentation? What is `m`? Presumably its short for `mode` (`fopen(3)` style), but we also have a `open(2)` style `mode`.  Maybe an enum for [read, write, append]?

Or maybe even Rust style functions:
- SCFopen: open for reading
- SCFcreate: open for writing and truncate